### DISCUSSION TITLE
Is it fine to use the function ( setup ) style?

### DISCUSSION THREAD

**Discussion by matthiashermsen** [vuejs/pinia#760] [Help and Questions] [answered]
Based on https://pinia.esm.dev/introduction.html#basic-example at 

> You can even use a function (similar to a component setup()) to define a Store for more advanced use cases

I'm using the function / "composition API" style and it works fine for me :) But all the examples in the docs use the object style approach and I would like to know if one approach is favored. The parts "even" and "more advanced use cases" might be confusing.

Is there a benefit in choosing the object ( Options API ) style instead of the function ( Composition API ) style?

Thanks in advance

**Comment by posva** [comment, marked as answer]
It's the same as Options API vs Composition API. Use whatever fits your need the best!

There are minor differences in:

- Devtools: Options Stores can group changes per actions while Setup Stores cannot
- Hydration: `hydrate()` vs `skipHydration()`
